sqlalchemy_bigquery/merge.py

```diff
@@ -0,0 +1,231 @@
+# Copyright 2021 Google LLC
+#
+# Use of this source code is governed by an MIT-style
+# license that can be found in the LICENSE file or at
+# https://opensource.org/licenses/MIT.
+
+import textwrap
+from copy import copy
+from dataclasses import dataclass
+from typing import Any, ClassVar, Dict, Optional, Tuple, Type, Union
+
+import sqlalchemy as sa
+from sqlalchemy.ext.compiler import compiles
+from sqlalchemy.sql.compiler import SQLCompiler
+from sqlalchemy.sql.expression import ClauseElement, ColumnElement
+from sqlalchemy.sql.selectable import Subquery  # noqa
+from sqlalchemy.sql.selectable import Alias, Select, TableClause
+
+MergeConditionType = Optional[ColumnElement[sa.Boolean]]
+
+
+class Merge(ClauseElement):
```

Comment:
`ClauseElement` instance is not executable, so passing a `Merge` instance to `Connection.execute()` doesn't work.

```python
updates = {"id": source.id, "name": source.name}
stmt = (
  Merge(target, source, on=target.id == source.id)
  .when_matched()
  .then_update(updates)
  .when_not_matched_by_target()
  .then_insert(updates)
)
connection.execute(stmt)
```

How about inheriting [`UpdateBase`](https://docs.sqlalchemy.org/en/14/core/dml.html#sqlalchemy.sql.expression.UpdateBase)?